src/main/java/teammates/ui/controller/InstructorFeedbackResultsPageData.java

```diff
@@ -25,92 +48,1262 @@
+            
+            sectionPanels.put(selectedSection, sectionPanel);
+        }
+    }
+
+    private void addMissingParticipantsForTeamToSectionPanelWithModerationButton(
+                                                             InstructorFeedbackResultsSectionPanel sectionPanel, 
+                                                             String teamName, List<String> teamMembers) {
+        for (String teamMember : teamMembers) {
+            InstructorResultsModerationButton moderationButton = buildModerationButtonForGiver(null, teamMember, 
+                                                                           "btn btn-default btn-xs",
+                                                                           "Moderate Responses");
+            InstructorFeedbackResultsGroupByQuestionPanel giverPanel = 
+                    buildInstructorFeedbackResultsGroupByQuestionPanel(teamMember, bundle.getFullNameFromRoster(teamMember),
+                                                                       new ArrayList<InstructorResultsQuestionTable>(), moderationButton, 
+                                                                       true);
+
+            giverPanel.setHasResponses(false);
+            addParticipantPanelToSectionPanel(sectionPanel, teamName, giverPanel);
+        }
+    }
+    
+    private void addMissingParticipantsForTeamToSectionPanelWithoutModerationButton(
+                                    InstructorFeedbackResultsSectionPanel sectionPanel, 
+                                    String teamName, List<String> teamMembers) {
+        for (String teamMember : teamMembers) {
+            InstructorFeedbackResultsGroupByQuestionPanel giverPanel = 
+            buildInstructorFeedbackResultsGroupByQuestionPanel(teamMember, bundle.getFullNameFromRoster(teamMember),
+                                                          new ArrayList<InstructorResultsQuestionTable>(), null, 
+                                                          false);
+            giverPanel.setHasResponses(false);
+            
+            addParticipantPanelToSectionPanel(sectionPanel, teamName, giverPanel);
+        }
+    }
+    
+
+    private void addParticipantPanelToSectionPanel(
+                                    InstructorFeedbackResultsSectionPanel sectionPanel, String currentTeam,
+                                    InstructorFeedbackResultsGroupByQuestionPanel giverPanel) {
+        List<InstructorResultsParticipantPanel> teamsMembersPanels;
+        
+        if (sectionPanel.getParticipantPanels().containsKey(currentTeam)) {
+            teamsMembersPanels = sectionPanel.getParticipantPanels().get(currentTeam);
+        } else {
+            teamsMembersPanels = new ArrayList<InstructorResultsParticipantPanel>();
+            sectionPanel.getParticipantPanels().put(currentTeam, teamsMembersPanels);
+        }
+        
+        teamsMembersPanels.add(giverPanel);
```

Comment:
While it is true that only the reference to `teamsMembersPanels` is stored in `sectionPanel`'s `participantPanels`, try to assume that passing in causes the object to be immutable and modify it only before passing it in.